addons/sale_project/models/account_move_line.py

```diff
@@ -15,7 +15,9 @@ def _compute_analytic_distribution(self):
         project_id = self._context.get('project_id', False)
         if project_id:
             project = self.env['project.project'].browse(project_id)
-            self.analytic_distribution = project._get_analytic_distribution()
+            for line in self:
+                if line.account_type not in ['asset_receivable', 'liability_payable']:
+                    line.analytic_distribution = project._get_analytic_distribution()
```

Comment:
instead of doing a for loop and call the method for each line, you could do something like this:
```py
lines = self.sorted(lambda line: line.account_type not in ['asset_receivable', 'liability_payable'])
lines.analytic_distribution = project._get_analytic_distribution()
```